src/geojson/geojson.ml

```diff
@@ -411,6 +425,7 @@ module Make (J : Intf.Json) = struct
             "A Geojson text should contain one object with a member `type`.")
 
   let to_json = function
+    | Feature f -> Feature.to_json ?foreign_members f
```

Comment:
```suggestion
```

This won't type-check because the types of the different branch statements are not of the same type. I think what we want to do is extended the `type t` to include `foreign_members` i.e. 

```ocaml
type geojson =
    | Feature of Feature.t
    | FeatureCollection of Feature.Collection.t
    | Geometry of Geometry.t

  and t = { geojson : geojson; bbox : float array option; foreign_members : (string * json) list option }
```

And use this to build the json, for example:

```ocaml
| { geojson = Feature f; bbox; foreign_members } -> Feature.to_json ?bbox ?foreign_members f
```